cli/src/pcluster/validators/scheduler_plugin_validators.py

```diff
@@ -148,3 +148,19 @@ def _validate(self, user_name):
                 ),
                 FailureLevel.ERROR,
             )
+
+
+class PluginInterfaceVersionValidator(Validator):
+    """Check if PluginInterfaceVersion is supported."""
+
+    def _validate(self, plugin_version, support_version):
+        low_limit, high_limit = packaging.version.parse(
```

Comment:
would be easer if we define the supported version range as a constant?
initially, it would be
```
SCHEDULER_PLUGIN_INTERFACE_VERSION_RANGE = ["1.0",  "1.0"]
```
or, if it's easier
```
SCHEDULER_PLUGIN_INTERFACE_VERSION_LOW_RANGE = "1.0"
SCHEDULER_PLUGIN_INTERFACE_VERSION_HIGH_RANGE = "1.0"
```

Then, we can easily compare with `plugin_version`
```
if not packaging.version.parse(SCHEDULER_PLUGIN_INTERFACE_VERSION_RANGE[0]) <= packaging.version.parse(plugin_version) <= packaging.version.parse(SCHEDULER_PLUGIN_INTERFACE_VERSION_RANGE[1])
  error
```

Where for the error you can keep the same logic to build the more appropriate error string, e.g. SCHEDULER_PLUGIN_INTERFACE_VERSION_RANGE[0] == SCHEDULER_PLUGIN_INTERFACE_VERSION_RANGE[1] etc...

Reply:
Then, for version 2, we can decide to drop or keep 1.0 backward compatibility. e.g.
* drop case
```
SCHEDULER_PLUGIN_INTERFACE_VERSION_RANGE = ["2.0",  "2.0"]
```
* backward case
SCHEDULER_PLUGIN_INTERFACE_VERSION_RANGE = ["1.0",  "2.0"]

Reply:
Changed